api/models/User.js

```diff
@@ -1,36 +1,39 @@
-const { cryptPassword } = require('../utils/encryption');
-const { Sequelize, Model, DataTypes } = require('sequelize');
-const sequelize = require('../db/db.js');
+const mongoose = require('mongoose');
+const bcrypt = require('bcrypt');
 
-class User extends Model {
-    constructor({ email, password }) {
-      super();
-      this.email = email;
-      this.password = password;
-    }
-
-    validPassword(password) {
-        return bcrypt.compare(password, this.password);
-    }
-}
+const saltRounds = 10;
 
-User.init({
-  email: DataTypes.STRING,
-  password: DataTypes.STRING,
-}, {
-  sequelize,
-  modelName: 'user'
+const UserSchema = new mongoose.Schema({
+  firstname: { type: String, required: true },
+  lastname: { type: String, required: true },
+  email: { type: String, required: true, unique: true },
+  password: { type: String, required: true }
 });
 
-User.beforeCreate((user, options) => {
-    return cryptPassword(user.password)
-        .then(success => {
-            user.password = success;
-        })
-        .catch(err => {
-            if (err) console.log(err);
-        });
+UserSchema.pre('save', function(next) {
+  if (this.isNew || this.isModified('password')) {
+    const document = this;
+    bcrypt.hash(this.password, saltRounds, function(err, hashedPassword) {
```

Comment:
The utitly `cryptPassword` from `require('../utils/encryption');` could also be used

Reply:
True... I will try to do it...